osdep/timer-linux.c

```diff
@@ -35,15 +41,24 @@ void mp_sleep_ns(int64_t ns)
 uint64_t mp_raw_time_ns(void)
 {
     struct timespec tp = {0};
-    int ret = -1;
-#if defined(CLOCK_MONOTONIC_RAW)
-    ret = clock_gettime(CLOCK_MONOTONIC_RAW, &tp);
-#endif
-    if (ret)
-        clock_gettime(CLOCK_MONOTONIC, &tp);
+    clock_gettime(clk_id, &tp);
     return MP_TIME_S_TO_NS(tp.tv_sec) + tp.tv_nsec;
 }
 
 void mp_raw_time_init(void)
 {
+    static const clockid_t clock_ids[] = {
+#if defined(CLOCK_MONOTONIC_RAW)
+        CLOCK_MONOTONIC_RAW,
+#endif
+        CLOCK_MONOTONIC,
+    };
+
+    struct timespec tp;
+    int ret = EINVAL;
+    for (int i = 0; i < MP_ARRAY_SIZE(clock_ids) && !ret; ++i) {
```

Comment:
I think you want to initialize ret as 0 here.

Reply:
late night changes, fixed